cpp/vcpkg.json

```diff
@@ -61,6 +61,7 @@
       "name": "arrow",
       "default-features": false
     },
+    "azure-core-cpp",
```

Comment:
Why don't we need this change in the Conda environment_unix.yml? Any changes to the Conda feedstock required?

Reply:
No need. Actually this line is not necessary as `azure-core-cpp` has already been depended by `azure-storage-blobs-cpp`.
I add this line because I need to specify the version of it in the later part of the file so I think it is less confusing to put it here as well